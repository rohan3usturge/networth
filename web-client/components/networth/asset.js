import { Amount } from "./amount";

const Asset = ({ asset, onChange, disabled, currency }) => {
  const handleChange = (changedAmount) => {
    const changed = { ...asset };
    changed.lineItem.amount = changedAmount;
    if (onChange) {
      onChange(changed);
    }
  };

  return (
    <tr className="d-flex">
      <td className="col-4">{asset.lineItem.name}</td>
      <td className="col-8">
        <div className="input-group mb-3">
          <Amount
            currency={currency}
            disabled={disabled}
            defaultValue={asset.lineItem.amount}
            currency={currency}
            className="form-control"
            onChange={handleChange}
          />
        </div>
      </td>
    </tr>
  );
};

export { Asset };
