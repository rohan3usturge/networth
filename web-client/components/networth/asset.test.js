import { shallow } from "enzyme";
import { Asset } from "./asset";

describe("Liability", () => {
  it("should render properly and match snapshot", () => {
    const wrapper = shallow(
      <Asset
        asset={{ meta: {}, lineItem: {} }}
        currency="CAD"
        disabled={true}
        onChange={() => {}}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
