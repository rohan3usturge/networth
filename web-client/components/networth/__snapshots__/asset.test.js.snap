// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Liability should render properly and match snapshot 1`] = `
<tr>
  <td />
  <td>
    <div
      className="input-group mb-3"
    >
      <Amount
        className="form-control"
        currency="CAD"
        disabled={true}
        onChange={[Function]}
      />
    </div>
  </td>
</tr>
`;
